generic.go

```diff
@@ -2,6 +2,28 @@ package pospop
 
 // count8 generic implementation
 func count8generic(counts *[8]int, buf []uint8) {
+	if uint64(len(buf)) >= 1<<(64-8) {
```

Comment:
The counter in `tmp[7]` is shifted up 7 bits so we can only represent a total up to 2 ^ (64 - 7) in the count. That means that if we get more the uint64 will overflow.

While in practice you would never see a slice that big I find it nice to have these checks instead of just assuming everything will be good.